### PR TITLE
fix: add s3 control to list of service clients that should have useSignedBodyHeader

### PR DESCRIPTION
**Pull request by dayaffe**
Similar to S3 and Glacier, S3ControlClient is required to send header `x-amz-content-sha256` indicating that the body is signed. 

[s3-control-customizations SEP](https://code.amazon.com/packages/AwsDrSeps/blobs/main/--/seps/accepted/shared/s3-control-customization.md) states ```Clients must also include the X-Amz-Content-Sha256 header in S3 Control API request signatures. The X-Amz-Content-Sha256 header is a required component of the s3v4 signature.```

## Issue \#
#659 

## Description of changes
- update list of IDs to include `"s3 control"` which support `useSignedBodyHeader` in `SigV4Config` instantiation
- include codegen changes to `S3ControlClient`
- update related unit tests for `SigV4Config`

This change was tested with passing unit tests and by running the code to replicate the issue as stated in #659 

## Conventional Commits

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/AWSSigningParams.kt [hunk 1]
```diff
@@ -52,7 +52,7 @@ class SigV4Configurator(
         useDoubleURIEncode = params.service.sdkId.lowercase(Locale.US) != "s3",
         useURLPathNormalization = params.service.sdkId.lowercase(Locale.US) != "s3",
         useUnsignedPayload = params.operation.hasTrait<UnsignedPayloadTrait>() || params.forceUnsignedBody,
-        useSignedBodyHeader = listOf("s3", "glacier").contains(params.service.sdkId.lowercase(Locale.US)) &&
+        useSignedBodyHeader = listOf("s3", "glacier", "s3 control").contains(params.service.sdkId.lowercase(Locale.US)) &&
```
Comment (jbelkins) [inline codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/AWSSigningParams.kt:55]: I can't remember off-hand, is sdkId the service identifier that is guaranteed not to change?
(I thought that one didn't have spaces in it)

Comment (dayaffe) [reply codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/AWSSigningParams.kt:55]: Yep it is! And it allows spaces

https://smithy.io/2.0/aws/aws-core.html#sdkid
- "The value MUST NOT change once a service is publicly released. If the value does change, the service will be considered a brand new service in the AWS SDKs and Tools."
- "... the value can only contain alphanumeric characters and spaces. ... and when using spaces, each space must be between two alphanumeric characters."

**Review by jbelkins — APPROVED** [review]

**Review by sichanyoo — APPROVED** [review]
Lgtm!

**Pull request merged**